ietfdata/tools/affiliations.py

```diff
@@ -0,0 +1,152 @@
+# TODO: Add licensing
+
+import sys
+import csv
+import json 
+import copy
+
+import datetime
+from datetime import timedelta
+from typing import Optional
+
+from ietfdata.datatracker     import *
+from ietfdata.datatracker_ext import *
+from ietfdata.mailarchive2    import *
+
+# load affiliation_mappings
+# import affiliations_map #hugo
+import ietfdata.tools.affiliation_mapping_dictionary as afmap #yangjun
+# affiliations.py --- script to generate extract affiliations
+# Mappings generated:
+# 1. raw affiliation -> normalised affiliation mappings
+# 2. email_domain -> normalised affiliations mappings
+# 3. identity -> start year-month, end year-month, affiliation mapping
+
+# Affiliation Entry Class
+class AffiliationEntry:
+    start_date  : datetime.date
+    end_date    : Optional[datetime.date]
+    names : list[str]
+    
+    def __init__(self, affiliation_str:str, start_date:datetime.date, end_date:Optional[datetime.date],names_list:Optional[list[str]]):
+        if names_list is not None:
+            self.names=copy.deepcopy(names_list)
+        else:
+            self.names = cleanup_affiliation(affiliation_str)
+        self.start_date = start_date
+        self.end_date = end_date
+    
+    def set_end_date(self, new_end_date:datetime.date):
+        self.end_date = new_end_date
+    
+    def set_affiliation_names(self, affiliation:str):
+        self.namse=cleanup_affiliation(affiliation)
+    
+    def match_names(self, names:list[str]):
+        for name in names:
+            if name not in self.names:
+                return False
+        return True
+    
+    def __str__(self):
+        return_str = '{"names":['
+        for name in self.names:
+            return_str+=f'"{name}",'
+        return_str =  return_str.rstrip(',')
+        return_str += "],"
+        return_str += f'"start_date":"{self.start_date}","end_date":"{self.end_date}"}}'
+        return return_str
+
+# Affiliation mapping class
+class AffiliationMap:
+    identifiers : list[str]
```

Comment:
What are the `identifiers` here? Is the goal is to represent mappings from a person to the different affiliations they had over time, in which case `identifiers` is presumably a set of names, email addresses, the Datatracker PersonURI, etc., that uniquely identify that person, or is the mapping something else? Likely the code doesn’t need to change, but a comment explaining the intended usage would help.